lib/ddtrace/buffer.rb

```diff
@@ -2,23 +2,77 @@
 require 'ddtrace/diagnostics/health'
 require 'ddtrace/runtime/object_space'
 
+# Trace buffer that accumulates traces for a consumer.
+# Consumption can happen from a different thread.
 module Datadog
-  # Trace buffer that stores application traces. The buffer has a maximum size and when
-  # the buffer is full, a random trace is discarded. This class is thread-safe and is used
-  # automatically by the ``Tracer`` instance when a ``Span`` is finished.
-  class TraceBuffer
-    def initialize(max_size)
-      @max_size = max_size
+  # Aggregate metrics:
+  # They reflect buffer activity since last #pop.
+  # These may not be as accurate or as granular, but they
+  # don't use as much network traffic as live stats.
+  class MeasuredBuffer
+    def initialize
+      @buffer_accepted = 0
+      @buffer_accepted_lengths = 0
```

Comment:
I'm not sure I understand the difference between `@buffer_accepted_lengths` and `@buffer_spans` ? Are one of these redundant?

Reply:
One is the number of traces pushed, the other is the total number of spans pushed. One tracer == one or more spans.

Reply:
if `@buffer_accepted_lengths` is the number of traces pushed, then i believe we shouldn't be subtracting by the trace.length in `measure_drop` and `measure_accept`, right? If i'm off base here feel free to disregard

https://github.com/DataDog/dd-trace-rb/blob/7511b140aa973bc99572ebd080b13d0cc9bd1ba4/lib/ddtrace/buffer.rb#L30-L31

Reply:
🐛 alert: ` @buffer_accepted_lengths -= trace.length `
Yeah, this is wrong, nice catch! It's an oldie too!

Reply:
I extracted this fix to a separate PR, https://github.com/DataDog/dd-trace-rb/pull/1182, as we'll likely want to report in separately in the release changelog.